manifests/init.pp

```diff
@@ -1,234 +1,111 @@
-# == Class: arc
+# @summary Manages tcl-devel package, libtcl symlink and /etc/rndrelease.
+#
+# This module assumes that you have OpenAFS installed. It is ment to be used in cooperation with
+# puppet-module-afs. It is needed for modulecmd to work properly on all platforms.
+#
+# @param manage_rndrelease
+#   Boolean to trigger management of /etc/rndrelease file.
+#
+# @param create_rndrelease
+#   Boolean to trigger creation of /etc/rndrelease file.
+#
+# @param create_symlink
+#   Boolean to trigger creation of libtcl symlink.
+#
+# @param install_package
+#   Boolean to trigger installation of packages.
+#
+# @param packages
+#   Array with package names to be installed.
+#
+# @param rndrelease_version
+#   String containing the content for /etc/rndrelease.
+#   If set to undef /etc/rndrelease will be deleted.
+#
+# @param symlink_target
+#   Absolute path to the target of the libtcl symlink.
+#
+# @param manage_arc_console_icon
+#   Boolean to trigger if arc_console.desktop should be managed.
+#
+# @param arc_console_icon
+#   Boolean to trigger creation (true) or deletion (false) of arc_console.desktop for the the arc_console.
 #
-# Manages tcl-devel package, libtcl symlink and /etc/rndrelease.
-
 class arc (
-  $manage_rndrelease       = true,
-  $create_rndrelease       = true,
-  $create_symlink          = true,
-  $install_package         = true,
-  $package_adminfile       = undef,
-  $packages                = 'USE_DEFAULTS',
-  $package_provider        = undef,
-  $package_source          = undef,
-  $rndrelease_version      = 'USE_DEFAULTS',
-  $symlink_target          = 'USE_DEFAULTS',
-  $manage_arc_console_icon = false,
-  $arc_console_icon        = false,
+  Boolean $manage_rndrelease = true,
+  Boolean $create_rndrelease = true,
+  Boolean $create_symlink = true,
+  Boolean $install_package = true,
+  Optional[Array[String[1]]] $packages = undef,
+  Optional[String[1]] $rndrelease_version = undef,
+  Optional[Stdlib::Absolutepath] $symlink_target = undef,
+  Boolean $manage_arc_console_icon = false,
+  Boolean $arc_console_icon = false,
 ) {
-
   # <define os default values>
   # Set $os_defaults_missing to true for unspecified osfamilies
 
-  case "${::operatingsystem}-${::operatingsystemrelease}" {
+  case "${facts['os']['name']}-${facts['os']['release']['full']}" {
```

Comment:
Wouldn't ${facts['os']['release']['major'] be better here since it is compared to major releases and not full release? This could prevent us from being forced to use regexp too, though that would require us to type each entry explicitly. Maybe better to keep as is.

Reply:
Lets continue this part in #41 then

Reply:
Oops yeah I missed it initially. Please see my response below. 

> Yes, shouldn't this be covered by lines 81-87?